mmv1/products/discoveryengine/DataStore.yaml

```diff
@@ -0,0 +1,168 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'DataStore'
+description: |
+  Vertex AI Search and Conversation data store is a collection of websites and
+  documents used to find answers for end-user's questions.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Create a search data store': 'https://cloud.google.com/generative-ai-app-builder/docs/create-data-store-es'
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1alpha/projects.locations.collections.dataStores'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores?dataStoreId={{data_store_id}}&createAdvancedSiteSearch={{create_advanced_site_search}}'
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}',
+  ]
+
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+
+examples:
```

Comment:
Unless you specifically want each of these examples in the website documentation and/or the resource is extremely cheap to provision, we probably don't need to test each configuration here. The most common pattern is one minimal example, and one example with every field specified.

Reply:
Thanks, I see.

Our resource DataStore has four types:
1. website
2. advanced_website
3. structured
4. unstructured
There are four test examples for each type respectively. The fields for each type are the same covering every field, only the value of some fields are different. 

I think it's okay to show the examples in the website documentation, because it is clear for the users to know the config settings for different types of DataStore. If only keeping one example for one type, do we need some instructions like README.md to specify the config setting for each type?

But I am not sure about  "extremely cheap" you mentioned. I checked the auto-generated code in terraform-provider-google, there is no difference between one test example and multiple test examples. Time used to run unit tests and acc tests may increase.

Reply:
Ok, makes sense. Feel free to add the other examples back if you do feel they provide value, but this pattern has the benefit of matching other resources more closely, and it makes tests faster to run (only really relevant for some types of resources that take multiple hours to provision or cost large amounts of money).

Reply:
Sure, thanks for your further explanation. Currently, being consistent and efficient sounds more reasonable, because we have API doc and error message will be returned if the user does not provide valid values. I can add other examples if needed in the future.